watertap/unit_models/electroNP_ZO.py

```diff
@@ -462,8 +462,18 @@ def calculate_scaling_factors(self):
         super().calculate_scaling_factors()
 
         iscale.set_scaling_factor(self.recovery_frac_mass_H2O, 1)
-        iscale.set_scaling_factor(self.energy_electric_flow_mass, 1e2)
-        iscale.set_scaling_factor(self.magnesium_chloride_dosage, 1e1)
+
+        if iscale.get_scaling_factor(self.energy_electric_flow_mass) is None:
+            sf = iscale.get_scaling_factor(
+                self.energy_electric_flow_mass, default=1e2, warning=True
+            )
+        iscale.set_scaling_factor(self.energy_electric_flow_mass, sf)
+
+        if iscale.get_scaling_factor(self.magnesium_chloride_dosage) is None:
+            sf = iscale.get_scaling_factor(
+                self.magnesium_chloride_dosage, default=1e1, warning=True
+            )
+        iscale.set_scaling_factor(self.magnesium_chloride_dosage, sf)
```

Comment:
Same here- this set_scaling_factor should be within the if conditional because we only want to set the scaling factor if the user has not provided one in their flowsheet. The way it is written now, if a user supplied a scaling factor for this variable and ran calculate_scaling_factors afterwards, the scaling factor they specified would be overwritten without them realizing.

Reply:
Also, just realized that you don't need both lines (get_scaling_factor _and_ set_scaling_factor), since the first will set a default scaling factor and show a warning.

Reply:
@adam-a-a I fixed the indent problem, but I am a little bit unsure why I don't need to `set_scaling_factor`, I checked other files like `nanofiltration_ZO.py`, they also do `get_scaling_factor` then `set_scaling_factor` even with a warning=True.